sycl/source/detail/scheduler/commands.hpp

```diff
@@ -290,6 +290,14 @@ class Command {
   // Memory is allocated in this method and released in destructor.
   void copySubmissionCodeLocation();
 
+  /// Clear all dependency events for device and host dependencies. This should
+  /// only be used if a command is about to be deleted without being executed
+  /// before that.
+  void clearAllDependencies() {
+    MPreparedDepsEvents.clear();
+    MPreparedHostDepsEvents.clear();
+  }
```

Comment:
I worry that just having this function opens up the chances of someone misusing it. Since it's only used in one place, could we maybe inline it and make it very clear there why it is valid?

Reply:
Thanks for the feedback @steffenlarsen!

I could not inline it at the place using it, because `MPreparedDepsEvents` and `MPreparedHostDepsEvents` are protected fields of the `Command` class, so I can't access them in `graph_builder.cpp`. 

The member methods for accessing them (`getPreparedHostDepsEvents` and `getPreparedDepsEvents`) both return a `const std::vector<...>`, so I can't `clear()` them from `graph_builder.cpp`. 

That was the reasoning to introduce `clearAllDependencies`. 

To avoid misuse, we can either extend the documentation to be very clear that this method is only intended for this use case, or we would need to drop the `const` qualifier from `getPrepared...` (which would open it to misuse in other places). 

What would you prefer?

Reply:
I see... In that case, I think I prefer keeping `clearAllDependencies`, but in that case I wonder if the name itself is a little misleading; is MDeps not also dependecies? I suppose `Node->MDeps.clear()` is only called in the same place we call this new member function anyway, so could we move `MDeps.clear()` in here too?

Reply:
Good point, I moved `MDeps.clear()` into `clearAllDependencies` and extended the documentation a bit.